kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -62,34 +62,32 @@ public class LogReplay {
         .add("protocol", Protocol.READ_SCHEMA)
         .add("metaData", Metadata.READ_SCHEMA);
 
+    /** We don't need to read the entire RemoveFile, only the path and dv info */
+    private static StructType REMOVE_FILE_SCHEMA = new StructType()
+        .add("path", StringType.STRING, false /* nullable */)
+        .add("deletionVector", DeletionVectorDescriptor.READ_SCHEMA, true /* nullable */);
+
+    private static StructType getAddSchema(boolean shouldReadStats) {
+        return shouldReadStats ? AddFile.SCHEMA_WITH_STATS :
+            AddFile.SCHEMA_WITHOUT_STATS;
+    }
+
     /**
-     * Read schema when searching for all the active AddFiles (need some RemoveFile info, too).
-     * Note that we don't need to read the entire RemoveFile, only the path and dv info.
+     * Read schema when searching for all the active AddFiles
      */
-    public static final StructType ADD_REMOVE_READ_SCHEMA = new StructType()
-        // TODO: further restrict the fields to read from AddFile depending upon
-        // the whether stats are needed or not: https://github.com/delta-io/delta/issues/1961
-        .add("add", AddFile.SCHEMA)
-        .add("remove", new StructType()
-            .add("path", StringType.STRING, false /* nullable */)
-            .add("deletionVector", DeletionVectorDescriptor.READ_SCHEMA, true /* nullable */)
-        );
-
-    public static int ADD_FILE_ORDINAL = ADD_REMOVE_READ_SCHEMA.indexOf("add");
-    public static int ADD_FILE_PATH_ORDINAL =
-        ((StructType) ADD_REMOVE_READ_SCHEMA.get("add").getDataType()).indexOf("path");
-    public static int ADD_FILE_DV_ORDINAL =
-        ((StructType) ADD_REMOVE_READ_SCHEMA.get("add").getDataType()).indexOf("deletionVector");
+    public static StructType getAddRemoveReadSchema(boolean shouldReadStats) {
+        return new StructType()
+            .add("add", getAddSchema(shouldReadStats))
+            .add("remove", REMOVE_FILE_SCHEMA);
+    }
 
-    public static int REMOVE_FILE_ORDINAL = ADD_REMOVE_READ_SCHEMA.indexOf("remove");
-    public static int REMOVE_FILE_PATH_ORDINAL =
-        ((StructType) ADD_REMOVE_READ_SCHEMA.get("remove").getDataType()).indexOf("path");
-    public static int REMOVE_FILE_DV_ORDINAL =
-        ((StructType) ADD_REMOVE_READ_SCHEMA.get("remove").getDataType()).indexOf("deletionVector");
+    public static int ADD_FILE_ORDINAL = 0;
```

Comment:
why remove the `ADD_REMOVE_READ_SCHEMA.indexOf("add")`? Same below.

Reply:
`ADD_REMOVE_READ_SCHEMA` is replaced by `getAddRemoveSchema(...)`

Reply:
I could make this `getAddRemoveSchema(false).indexOf("add")` if you'd prefer

Reply:
I think this is ok.